### ISSUE TITLE
Simplify db-sync

### ISSUE THREAD

**Issue by kderme** [input-output-hk/cardano-db-sync#813]
db-sync code is sometimes hard to follow because it has different interfaces, datalayer, and plugins.
We could simplify these:
- Since smash is getting integrated to db-sync, there is probably no need for the data layer and the separation between cardano-sync and cardano-db-sync
- We could integrate db-sync and db-sync-extended executables with a simple option and remove the plugin interfaces

Also db-sync extracts and handles a lot of data from blockchain and the ledger state, making it heavy and slow. We could provide different flags which make db-sync ignore parts of these data. For example a flag which disables the smash functionality or the Alonzo scripts related tables. On the other hand someone may be interested in a db-sync flavor which only focuses on these things: smash or scripts related tables.

**Comment by erikd** [comment]
I completely agree with both main points. In fact the first is a duplicate of #598 .

However this:

> We could provide different flags which make db-sync ignore parts of these data.

could easily make both development and testing far more difficult than they should be.

**Comment by kderme** [comment]
> could easily make both development and testing far more difficult than they should be.

It could also help development though. The whole debugging iteration in mainnet is a huge pain, because of the slow sync times. Ignoring some of the data could make it faster.

**Comment by kderme** [comment]
First 2 bullets are done. Last paragraph is transfered to https://github.com/input-output-hk/cardano-db-sync/issues/1195
